packages/iota/src/validation/outputs/common.ts

```diff
@@ -0,0 +1,325 @@
+    }
+
+    return mergeValidationResults(...results);
+}
+
+/**
+ * Validate the amount is greater than zero.
+ * @param amount The amount to validate.
+ * @param outputName The name of the output to use in the error message.
+ * @returns The validation result.
+ */
+function validateAmountIsGreaterThanZero(amount: string, outputName: string): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    if (bigInt(amount).leq(bigInt.zero)) {
+        result = failValidation(result, `${outputName} output amount field must be larger than zero.`);
+    }
+
+    return result;
+}
+
+/**
+ * Validate the amount is lesser than maximum token supply.
+ * @param amount The amount to validate.
+ * @param tokenSupply The tokken supply amount.
+ * @param outputName The name of the output to use in the error message.
+ * @returns The validation result.
+ */
+function validateAmountIsLesserThanMaxSupply(
+    amount: string,
+    tokenSupply: string,
+    outputName: string
+): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    if (bigInt(amount).gt(tokenSupply)) {
+        result = failValidation(result, `${outputName} output amount field must not be larger than max token supply.`);
+    }
+
+    return result;
+}
+
+/**
+ * Validate the count is within allowed boundries.
+ * @param count The number to validate.
+ * @param min The maximum allowed value.
+ * @param max The maximum allowed value.
+ * @param outputName The name of the output to use in the error message.
+ * @param elementName The name of the validation subject to use in the error message.
+ * @returns The validation result.
+ */
+function validateCount(
+    count: number,
+    min: number,
+    max: number,
+    outputName: string,
+    elementName: string
+): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    if (count < min || count > max) {
+        const message = min === max ?
+            `${outputName} output ${elementName} count must be equal to ${max}.` :
+            `${outputName} output ${elementName} count must be between ${min} and ${max}.`;
+
+        result = failValidation(result, message);
+    }
+
+    return result;
+}
+
+/**
+ * Validate that the unlock conditions define only allowed types.
+ * @param outputType The type of the output.
+ * @param unlockConditions The amount to validate.
+ * @param outputName The name of the output to use in the error message.
+ * @returns The validation result.
+ */
+function validateUnlockConditionAllowedTypes(
+    outputType: number,
+    unlockConditions: UnlockConditionTypes[],
+    outputName: string
+): IValidationResult {
+    let result: IValidationResult = { isValid: true };
+
+    const allowedUnlockConditionTypes = OUTPUT_TYPE_TO_SUPPORTED_PROP_VALUES.get(outputType)!.unlockConditions;
+
+    if (!unlockConditions.every(uC => allowedUnlockConditionTypes.includes(uC.type))) {
+        const unlockConditionNames = UNLOCK_CONDITION_TYPE_NAMES
```

Comment:
Please
```
UNLOCK_CONDITION_TYPE_NAMES.filter(
    (uC, index) => allowedUnlockConditionTypes.includes(index)
);
```
if lines need to be shortened. Easier to read/edit this way. No random "enter" slapping.

Reply:
Sure will refactor, but the enter was not randomly slapped rather filter method was pushed to a new line in the similar fashion as when you chain multiple method calls on a single object.